packages/core/event.js

```diff
@@ -1,41 +1,49 @@
 const ErrorStackParser = require('./lib/error-stack-parser')
 const StackGenerator = require('stack-generator')
 const hasStack = require('./lib/has-stack')
-const { reduce, filter } = require('./lib/es-utils')
+const { map, reduce, filter } = require('./lib/es-utils')
 const jsRuntime = require('./lib/js-runtime')
 const metadataDelegate = require('./lib/metadata-delegate')
 const isError = require('./lib/iserror')
 
 class Event {
   constructor (errorClass, errorMessage, stacktrace = [], handledState = defaultHandledState(), originalError) {
-    // private (un)handled state
-    this._handledState = handledState
-
-    // setable props
-    this.app = undefined
     this.apiKey = undefined
-    this.breadcrumbs = []
     this.context = undefined
-    this.device = undefined
-    this.errorClass = ensureString(errorClass)
-    this.errorMessage = ensureString(errorMessage)
     this.groupingHash = undefined
-    this._metadata = {}
-    this.request = undefined
+    this.originalError = originalError
+
+    this._handledState = handledState
     this.severity = this._handledState.severity
-    this.stacktrace = reduce(stacktrace, (accum, frame) => {
-      const f = formatStackframe(frame)
-      // don't include a stackframe if none of its properties are defined
-      try {
-        if (JSON.stringify(f) === '{}') return accum
-        return accum.concat(f)
-      } catch (e) {
-        return accum
-      }
-    }, [])
+    this.unhandled = this._handledState.unhandled
+
+    this.app = {}
+    this.device = {}
+    this.request = {}
+
+    this.breadcrumbs = []
+
+    this._metadata = {}
     this._user = {}
-    this.session = undefined
-    this.originalError = originalError
+    this._session = undefined
+
+    this.errors = [
+      {
+        errorClass: ensureString(errorClass),
+        errorMessage: ensureString(errorMessage),
+        type: jsRuntime,
```

Comment:
Should this be defined on `interface Error` or is it an internal thing?

Just wondering if a plugin wanted to add a second error to this event would it need to remember to add  that `type: jsRuntime` or can we avoid that complexity by moving it to the `exceptions` projection in `toJSON`?

Reply:
It's public and should be on the `Error` type so I'm adding it. In theory we can have a js error which is "caused by" a native error, and both may be in the array.